### ISSUE TITLE
performance - option to use N worker threads in initWorker

### ISSUE THREAD

**Issue by tomholub** [openpgpjs/openpgpjs#641]
I've done some informal performance testing (involving our own code + OpenPGP.js) to see the difference between using a single worker thread vs 8 threads with a round-robin.

The test creates 20 or 50 iframes with that load and decrypt a short message, with 1 or 8 workers. This mimics opening an email thread. Results in seconds to decrypt all (lower is better):

```
20 msgs
1 worker	9.31	9.20	9.17
8 workers	7.09	6.42	6.35

50 msgs
1 worker	33.51	24.12	24.13
8 workers	15.25	15.44	15.41
```

I think it would be worth to allow N worker threads setting for initWorker.

For the tests, I have been following this pattern:
```js
function initWorker() {
  if (worker || typeof window !== 'undefined' && window.Worker) {
    asyncProxy = [
      new _async_proxy2.default({ path: path, worker: worker, config: _config2.default }),
      new _async_proxy2.default({ path: path, worker: worker, config: _config2.default }),
      ... 
    ];
    return true;
  }
}

// round robin worker select
function getWorker() {
    if(asyncProxy) {
        asyncProxyLast = (asyncProxyLast + 1) % asyncProxy.length;
        return asyncProxy[asyncProxyLast];
    }
  return asyncProxy;
}

// always use getWorker() instead of direct reference when using, eg:
  if (asyncProxy) {
    return getWorker().delegate('reformatKey', options);
  }
```

This is just to give you a rough idea, it would probably be better to use `asyncProxies = []` and test for zero length + other cleanups.

**Comment by tomholub** [comment]
To clarify, the 1 or 8 workers were shared across all iframes: decryption was happening in a background script. That's why having a single worker becomes a bottleneck, and having an option to init N workers improves performance in a noticeable way.

**Comment by bartbutler** [comment]
Agreed, this is a good idea. I wonder if multiple workers utilize multiple cores or if all workers run in the same thread. The answer is probably browser-dependent.

**Comment by tomholub** [comment]
To my knowledge, they are supposed to utilize separate threads. Web worker
docs say that each web worker spawns an actual separate process which may
run in another core. At the same time, I did not see a 4x improvement in
speed (got 4 cores), so also unsure. There was a noticeable improvement
though.

**Comment by bartbutler** [comment]
Separate process may still have been sharing cores. In any case, this is a good idea, we would welcome a PR, otherwise I'll try to get to it when I can.

**Comment by tomholub** [comment]
I'll do a PR within a month

**Comment by sanjanarajan** [comment]
Closed by #661
